src/front/app/features/challenge/components/Challenge.js

```diff
@@ -13,12 +12,16 @@ class Challenge extends Component {
         props.getChallenge(id);
     }
 
+    componentWillUnmount() {
+        this.props.getInitialState();
+    }
+
     render() {
         /* istanbul ignore next */
         return (
             <div className="challenge-details">
                 <ChallengeNav />
-                <ChallengeDetails {...this.props.challenge.toJS()} />
+                {this.props.challenge.size && <ChallengeDetails challenge={this.props.challenge} />}
```

Comment:
I would show a preloader/progress bar in case threre're no challenge, but we can left it for the future.